### PR TITLE
Update update readme, add development, roadmap and contributing docs

### PR DESCRIPTION
**Pull request by cdrage**
- Updates the README
- Adds a development document
- Adds a contributing document
- Moves the "todo" to a ROADMAP

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+* Make your changes in a new git branch:
+
+     ```shell
+     git checkout -b bug/my-fix-branch master
+     ```
+
+* Create your patch, **including appropriate test cases**.
+* Include documentation that either describe a change to a behavior of kapp or the changed capability to an end user of kapp.
+* Commit your changes using **a descriptive commit message**. If you are fixing an issue please include something like 'this closes issue #xyz'.
+* Make sure your tests pass! As we use [travis-ci](https://travis-ci.org/) with `make test` it's recommended to run the command locally so that it will pass in the CI.
```
Comment (surajssd) [inline CONTRIBUTING.md]: not sure about travis yet!

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    ```
+
+That's it! Thank you for your contribution!
+
+### Merge Rules
+
+* Include unit or integration tests for the capability you have implemented
+* Include documentation for the capability you have implemented
+* If you are fixing an issue within Kapp.
+ * include the issue number you are fixing
```
Comment (surajssd) [inline CONTRIBUTING.md]: first white space `s/ /` and `s/i/I`

**Comment by surajssd** [comment]
Apart from the nit this LGTM! Awesome thing @cdrage

**Comment by cdrage** [comment]
@surajssd done!

**Comment by surajssd** [comment]
@cdrage thanks a ton merging this one! :+1:

File: README.md [hunk 3]
```diff
@@ -1,68 +1,80 @@
-# Kapp (Kubernetes + App)
+# Kapp - Openly Compose Applications for Kubernetes
```
Comment (ashetty1) [inline README.md]: The 'openly compose ...' part doesn't sound apt.

**Pull request merged**